### PR TITLE
[ie/youtube] Skip iOS formats that require PO Token

### PR DESCRIPTION
**Pull request by coletdjnz**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

Fix for one of the issues in https://github.com/yt-dlp/yt-dlp/issues/11868

It appears YouTube is rolling out what we presume is a PO Token requirement on iOS. It has the same characteristics as android: either immediate fail or fails part way through the download, and no StreamProtectionStatus is provided in the UMP/SABR responses.

iOS HLS formats appear to still work without PO Token.

This PR marks the iOS client as requiring a PO Token (except for HLS formats).
- Formats missing a PO Token that require one are **skipped** by default, unless `--extractor-args youtube:formats=missing_pot` is provided. 
- Improve warnings/logs around missing PO Token

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [X] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [ ] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [X] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [X] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

File: yt_dlp/extractor/youtube.py [hunk 1]
```diff
@@ -3973,13 +3974,10 @@ def append_client(*client_names):
...
             require_po_token = self._get_default_ytcfg(client).get('REQUIRE_PO_TOKEN')
-            if not po_token and require_po_token:
+            if not po_token and require_po_token and 'missing_pot' in self._configuration_arg('formats'):
                 self.report_warning(
                     f'No PO Token provided for {client} client, '
-                    f'which is required for working {client} formats. '
-                    f'You can manually pass a PO Token for this client with '
-                    f'--extractor-args "youtube:po_token={client}+XXX"',
-                    only_once=True)
+                    f'which may be required for working {client} formats. This client will be deprioritized.', only_once=True)
```
Comment (bashonly) [inline yt_dlp/extractor/youtube.py:3980]:
```suggestion
                    f'which may be required for working {client} formats. This client will be deprioritized', only_once=True)
```

File: yt_dlp/extractor/youtube.py [hunk 2]
```diff
@@ -3973,13 +3974,10 @@ def append_client(*client_names):
             )
 
             require_po_token = self._get_default_ytcfg(client).get('REQUIRE_PO_TOKEN')
-            if not po_token and require_po_token:
+            if not po_token and require_po_token and 'missing_pot' in self._configuration_arg('formats'):
```
Comment (bashonly) [inline yt_dlp/extractor/youtube.py:3977]: the formats are still deprioritized even if `missing_pot` is passed, though, right? we are still setting `'preference': -20`

File: yt_dlp/extractor/youtube.py [hunk 3]
```diff
@@ -4053,6 +4051,21 @@ def _needs_live_processing(self, live_status, duration):
                 or (live_status == 'post_live' and (duration or 0) > 2 * 3600)):
             return live_status
 
+    def _report_pot_format_skipped(self, video_id, client_name, proto):
+        msg = (
+            f'{video_id}: {client_name} client {proto} formats require a PO Token which was not provided. '
+            'They will be skipped as they may yield HTTP Error 403. '
+            f'You can manually pass a PO Token for this client with --extractor-args "youtube:po_token={client_name}+XXX. '
+            'For more information, refer to  https://github.com/yt-dlp/yt-dlp/wiki/Extractors#po-token-guide . '
+            'To enable these broken formats anyway, pass --extractor-args "youtube:formats=missing_pot".')
```
Comment (bashonly) [inline yt_dlp/extractor/youtube.py:4060]:
```suggestion
            'To enable these broken formats anyway, pass --extractor-args "youtube:formats=missing_pot"')
```

File: yt_dlp/extractor/youtube.py (hunk 2 above)
Comment (coletdjnz) [reply yt_dlp/extractor/youtube.py:3977]: yes

Comment (bashonly) [reply yt_dlp/extractor/youtube.py:3977]: disregard

**Review by bashonly — APPROVED** [review]

File: yt_dlp/extractor/youtube.py (hunk 2 above)
Comment (tcely) [reply yt_dlp/extractor/youtube.py:3977]: As I understand this code:

- `missing_pot` wasn't used: the clients requiring pot are skipped. The preference doesn't matter?
- this case, `missing_pot` was used: `deprioritize_pr` is set to `True` and the preference is `-20`

Comment (coletdjnz) [reply yt_dlp/extractor/youtube.py:3977]: > missing_pot wasn't used: the clients requiring pot are skipped. The preference doesn't matter?

The client is not skipped, but the formats requiring PO Token are. 

> this case, missing_pot was used: deprioritize_pr is set to True and the preference is -20

Correct

**Pull request merged**